### PR TITLE
fix(amf): fix for the issue 10925

### LINKED ISSUE #10925: [Regression-Failure] TC14b failing as AMF is not receiving response from Subscriber-db .
Your Environment

- Version:  1.7.0-1639678730-75fa541b  
- Affected Component: Access Gateway 
- Affected Subcomponent : AGW service
- Deployment Environment: Vagrant (AGW)

### Describe the Issue
TC14b failing as AMF is not receiving response from Subscriber-db , for the auth-info request for the UE from subscriber-db.

To Reproduce
1. AGW should be UP, and all services should be active and running.
2. All connection Procedures should be successfull.
3. Trigger the Regression SUIT and see the output of below TC.

4. TC failing as there is no Auth-Request sent to UE as AMF is not getting response from subscriber-db for the auth-info request sent.

**Expected behavior**
Registration procedure should get triggered as per testcase expectation.

**Screenshots**
MME_Logs_Failure:

MME_Logs_SUccess Flow:

Syslogs_Failure_Logs:

[https://app.zenhub.com/files/170803235/9e64d055-2a75-4bad-9a9f-d0d7014ac6d8/download](https://app.zenhub.com/files/170803235/9e64d055-2a75-4bad-9a9f-d0d7014ac6d8/download)

### PR DESCRIPTION
**Pull request by VinayKumar1591**
fix(amf): fix for the issue 10925

## Summary

1. Changes to send registration reject incase no response from subscriberDB.

## Test Plan
1. Verified sanity on TVM
2

### CODE REVIEW

File: lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp [hunk 1]
```diff
@@ -820,4 +820,29 @@ TEST_F(AMFAppProcedureTest, TestRegistrationProcSUCIExt) {
...
+      amf_app_desc_p, 36, 1, 1, 0, plmn, initial_ue_message_hexbuf,
+      sizeof(initial_ue_message_hexbuf));
+
+  /* Check if UE Context is created with correct imsi */
+  EXPECT_TRUE(get_ue_id_from_imsi(amf_app_desc_p, imsi64, &ue_id));
+
+  /* Send the authentication response message from subscriberdb */
+  int rc = RETURNok;
+  rc     = send_proc_authentication_info_answer(imsi, ue_id, false);
+  EXPECT_TRUE(rc == RETURNok);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:846]: should not you be checking the UE context post failure?

**Review by ulaskozat — APPROVED** [review]
LGTM with a nit comment. Mark as ready to merge once it is addressed.

**Review by ulaskozat — CHANGES_REQUESTED** [review]
actually there is a unit test failure due to memory leak. please fix it.

File: lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp (hunk 1 above)
Comment (VinayKumar1591) [reply lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:846]: UE Context gets freed in this case as it is negative scenario

**Comment by VinayKumar1591** [comment]
> actually there is a unit test failure due to memory leak. please fix it.

memory leaks fixed

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by ulaskozat** [comment]
@VinayKumar1591 fix the lint error.

**Comment by VinayKumar1591** [comment]
> @VinayKumar1591 fix the lint error.

Done

File: lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp (hunk 1 above)
Comment (ulaskozat) [reply lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:846]: yes, what I meant was that you should check whether the context got deleted or not...

Comment (VinayKumar1591) [reply lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:846]: added check whether ue context deleted or not

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**